### PR TITLE
Added a timeout condition which will throw a timeout error whenever the server of backplane-api is down

### PR DESCRIPTION
**Pull request by Dee-6777**
### What type of PR is this?

_feature_

### What this PR does / Why we need it?

This PR will throw a timeout error whenever the backplane-api server is down for 10 seconds or more

### Which Jira/Github issue(s) does this PR fix?

_Resolves #[OSD-17197](https://issues.redhat.com/browse/OSD-17197)_

### Special notes for your reviewer

You can test it locally by decreasing the timeout duration to a few milliseconds. It will throw a timeout error.
### Pre-checks (if applicable)

- [X] Ran unit tests locally
- [X] Validated the changes in a cluster
- [ ] Included documentation changes with PR

### CODE REVIEW

File: internal/github/github.go [hunk 1]
```diff
@@ -38,7 +39,7 @@ func NewClient(opts ...ClientOption) *Client {
 
 // GetLatestVersion returns latest version from the github API
 func (c *Client) GetLatestVersion(ctx context.Context) (latest upgrade.Release, err error) {
-	ctx, cancel := context.WithCancel(ctx)
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
```
Comment (samanthajayasinghe) [inline internal/github/github.go:42]: pls add a unit test to cover (>10s and < 10s) `timeout` condition

**Review by hectorakemp — APPROVED** [review]

**Pull request merged**